client/reducers/crudReducer.js

```diff
@@ -0,0 +1,61 @@
+import * as ActionTypes from '../actions/actionTypes';
+
+export default (state = {}, action) => {
+    switch (action.type) {
+
+        // Create user
+        case ActionTypes.CREATE_USER_SUCCESS:
+            return {
+                error: false,
+                message: 'User was successfully created'
+            };
+        
+        case ActionTypes.CREATE_USER_FAILURE:
+            return {
+                error: true,
+                message: action.payload
+            };
+
+        // Update user
+        case ActionTypes.UPDATE_USER_SUCCESS:
+            return {
+                error: false,
+                message: 'User was successfully updated'
+            };
+        
+        case ActionTypes.UPDATE_USER_FAILURE:
+            return {
+                error: true,
+                message: action.payload
+            };
+
+        // Admin create user
+        case ActionTypes.ADMIN_CREATE_USER_SUCCESS:
+            return {
+                error: false,
+                message: 'User was successfully created'
+            };
+        
+        case ActionTypes.ADMIN_CREATE_USER_FAILURE:
+            return {
+                error: true,
+                message: action.payload
+            };
+
+        // Delete user
+        case ActionTypes.DELETE_USER_SUCCESS:
+            return {
+                error: false,
+                message: 'User was successfully deleted'
+            };
+        
+        case ActionTypes.DELETE_USER_FAILURE:
+            return {
+                error: true,
+                message: action.payload
+            };
+
+        default:
+            return state;
+    }
+}
```

Comment:
This is the crud reducer, it looks big, but it's very simple - the actions coming in are all the same, they just correspond to a different CRUD element. Every action is either a success (in which case we indicate that there are no errors) or a failure (in which case we keep track of the error message). Essentially, this is only useful for displaying those error messages to the user - example: "Invalid password" or "The email doesn't exist", etc.

Reply:
Wonder if there could be a better name than crud reducer.